hy/core/macros.hy

```diff
@@ -20,6 +20,16 @@
          ~@(interleave (repeat name) rest))
      ~name))
 
+
+(defmacro assoc [coll k1 v1 &rest other-kvs]
+  (if (odd? (len other-kvs))
+    (macro-error (last other-kvs) "`assoc` takes an odd number of arguments"))
+  (setv c (if other-kvs (gensym "c") coll))
+  `(setv ~@(+ (if other-kvs [c coll] []) #* (genexpr
+    [`(get ~c ~k) v]
+    [[k v] (partition (+ (, k1 v1) other-kvs))]))))
```

Comment:
This is not indented properly, which makes it hard to read. Emacs can do this for you.